components/influxdb/scripts/create.py

```diff
@@ -28,31 +28,50 @@ def _configure_influxdb(host, port):
     # curl -S -s "http://localhost:8086/db?u=root&p=root" '-d "{\"name\": \"cloudify\"}"  # NOQA
     import urllib
     import urllib2
+    import ast
 
-    endpoint = 'http://{0}:{1}/db'.format(host, port)
+    endpoint_for_list = 'http://{0}:{1}/db'.format(host, port)
+    endpoint_for_creation = 'http://{0}:{1}/cluster/database_configs/' \
+                            '{2}'.format(host, port, db_name)
     params = urllib.urlencode(dict(u=db_user, p=db_pass))
-    data = {'name': db_name}
-    url = endpoint + '?' + params
+    url_for_list = endpoint_for_list + '?' + params
+    url_for_creation = endpoint_for_creation + '?' + params
 
     # check if db already exists
-    db_list = eval(urllib2.urlopen(urllib2.Request(url)).read())
+    db_list = eval(urllib2.urlopen(urllib2.Request(url_for_list)).read())
     try:
         assert not any(d.get('name') == db_name for d in db_list)
     except AssertionError:
         ctx.logger.info('Database {0} already exists!'.format(db_name))
         return
 
-    ctx.logger.info('Request is: {0} \'{1}\''.format(url, data))
-
     try:
-        urllib2.urlopen(urllib2.Request(url, json.dumps(data)))
+        utils.deploy_blueprint_resource(
+            '{0}/retention.json'.format(CONFIG_PATH),
+            '/tmp/retention.json', INFLUX_SERVICE_NAME)
+        with open('/tmp/retention.json') as f:
+            retention_policy = f.read()
+        ctx.logger.info(
+            'using retention policy: \n{0}'.format(retention_policy))
+        data = json.dumps(ast.literal_eval(retention_policy))
+        ctx.logger.info(
+            'using retention policy: \n{0}'.format(data))
+        cont_len = len(data)
+        req = urllib2.Request(url_for_creation, data, {
+            'Content-Type': 'application/json', 'Content-Length': cont_len})
+        ctx.logger.info('Request is: {0}'.format(req))
+        f = urllib2.urlopen(req)
+        response = f.read()
+        ctx.logger.debug(response)
```

Comment:
If the response is not as expected for some reason, there's no context to this being a response. This should be

```python
ctx.logger.info('Response: {0}'.format(response))
```